### PR TITLE
Fix ordered by

### PR DESCRIPTION
**Pull request by cedvdb**
Make tests introduced in https://github.com/atn832/fake_cloud_firestore/pull/215 pass.

### CODE REVIEW

File: test/mock_query_test.dart [hunk 1]
```diff
@@ -1249,24 +1379,32 @@ void main() {
...
-        .withConverter(fromFirestore: from, toFirestore: to);
-    await moviesCollection.add(Movie()..title = 'A long time ago');
-    await moviesCollection.add(Movie()..title = 'Robot from the future');
-    final searchResults =
-        await moviesCollection.orderBy('title').startAt(['R']).get();
-    expect(searchResults.size, equals(1));
-    final movieFound = searchResults.docs.first.data();
-    expect(movieFound.title, equals('Robot from the future'));
-  });
+  // test('startAt with converters', () async {
```
Comment (atn832) [inline test/mock_query_test.dart:1393]: Did you forget to uncomment?

File: lib/src/mock_query.dart [hunk 2]
```diff
@@ -71,52 +71,100 @@ class MockQuery<T extends Object?> extends FakeQueryWithParent<T> {
...
+          final searchedValue = values[i];
+          var index = 1 +
+              sublist.lastIndexWhere((doc) {
+                if (doc.data() == null) {
+                  return false;
+                }
+                final docValue = doc.get(keyName);
+                return docValue.compareTo(searchedValue) <= 0;
+              });
+          sublist = sublist.sublist(0, index);
```
Comment (atn832) [inline lib/src/mock_query.dart:163]: I haven't checked in detail yet, but this code looks very similar to _cursorUtil. Would you mind pointing out what makes this code necessary, as opposed to reusing _cursorUtil?

File: test/mock_query_test.dart (hunk 1 above)
Comment (cedvdb) [reply test/mock_query_test.dart:1393]: yes

**Comment by atn832** [comment]
Thanks for making the tests pass! I'll clean up the duplicate code introduced in this PR later.

**Comment by atn832** [comment]
Just published your fix to https://pub.dev/packages/fake_cloud_firestore/changelog#123. Thank you @cedvdb!

**Pull request merged**